Sources/_StringProcessing/Engine/MEBuiltins.swift

```diff
@@ -34,13 +34,21 @@ extension Processor {
     _ isStrictASCII: Bool,
     _ isScalarSemantics: Bool
   ) -> Input.Index? {
+
+    // ASCII fast-path
+    if let (next, result) = input._quickMatch(
+      cc, at: currentPosition, isScalarSemantics: isScalarSemantics
+    ) {
+      return result == isInverted ? nil : next
```

Comment:
Assert that the result is the same as the slow path, by refactoring the slow path into a separate function.